plugins/module_utils/foreman_helper.py

```diff
@@ -433,13 +461,16 @@ def __init__(self, argument_spec, **kwargs):
 
 class ForemanEntityApypieAnsibleModule(ForemanApypieAnsibleModule):
 
-    def __init__(self, argument_spec, **kwargs):
+    def __init__(self, argument_spec=None, **kwargs):
+        entity_spec, gen_args = _entity_spec_helper(kwargs.pop('entity_spec', {}))
+        if argument_spec is None:
```

Comment:
Perhaps this should be lower and reversed?
```python
if argument_spec is not None:
    args.update(argument_spec)
```

Reply:
you mean without changing the logic, we can scrap a few lines of code?
Maybe yes. About beeing reversed, i think whatever comes through the `argument_spec` parameter to `__init__` should have the highest priority.

Reply:
Yep, this is what I meant.